### ISSUE TITLE
Error in Pipeline Create-Container.ps1 when using artifact URL

### ISSUE THREAD

**Issue by GreatScott000** [microsoft/navcontainerhelper#1100]
**Describe the issue**
Getting the following error in the "Create Build Container" step when using a specific artifact URL ("https://bcartifacts.azureedge.net/onprem/14.8.38658.0/gb", which works in a standalone NAVContainerHelper script.

> No such object: onprem:onprem-14.8.38658.0-gb

**Scripts used to create container and cause the issue**
unmodified create-container.ps1

Attached here is the content of the raw log from the pipeline.
[Create Build Container Error.txt](https://github.com/microsoft/navcontainerhelper/files/4903211/Create.Build.Container.Error.txt)

**Full output of scripts**
```
Starting: Create Build Container
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : 2.170.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
Formatted command: . 'C:\EvolveiMS\Agent\_work\3\s\scripts\Create-Container.ps1'
========================== Starting Command Output ===========================
"C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\EvolveiMS\Agent\_work\_temp\7a5edaea-3b7a-433a-8290-b4313c043a30.ps1'"
Create onprem-5CG70948WR from https://bcartifacts.azureedge.net/onprem/14.8.38658.0/gb
WARNING: Container name should not exceed 15 characters
NavContainerHelper is version 0.7.0.13
NavContainerHelper is running as administrator
Host is Microsoft  - 1909
Docker Client Version is 19.03.8
Docker Server Version is 19.03.8
Removing C:\ProgramData\NavContainerHelper\Extensions\onprem-5CG70948WR
ArtifactUrl and ImageName specified
##[error]Error: No such object: onprem:onprem-14.8.38658.0-gb
Image onprem:onprem-14.8.38658.0-gb already exists
Building image onprem:onprem-14.8.38658.0-gb based on https://bcartifacts.azureedge.net/onprem/14.8.38658.0/gb
Generic Tag: 0.1.0.6
Container OS Version: 10.0.18363.900 (1909)
Host OS Version: 10.0.18363.900 (1909)
Copying Platform Artifacts
Copy Database
Copy Licensefile
Copy Installers
Copy TestToolKit
Copy UpgradeToolKit
Copy Extensions
c:\$TMP$-onprem-onprem-14.8.38658.0-gb
Sending build context to Docker daemon  1.114GB

Step 1/6 : FROM mcr.microsoft.com/dynamicsnav:10.0.18363.900-generic
 ---> 9353f4c5f346
Step 2/6 : ENV DatabaseServer=localhost DatabaseInstance=SQLEXPRESS DatabaseName=CRONUS IsBcSandbox=N artifactUrl=https://bcartifacts.azureedge.net/onprem/14.8.38658.0/gb
 ---> Using cache
 ---> cc91e96dda22
Step 3/6 : COPY my /run/
 ---> Using cache
 ---> 3e89a9e7015f
Step 4/6 : COPY NAVDVD /NAVDVD/
 ---> 28b5cdeb83a0
Step 5/6 : RUN \Run\start.ps1 -installOnly
```

**Screenshots**

**Additional context**

**Comment by freddydk** [comment]
Please include the script even though it is unmodified, there have been several versions and I need to know which you are running.

**Comment by GreatScott000** [comment]
Sorry Freddy.  I thought I attached it last time but forgot I can't drag/drop ps1 files.  I have renamed it to txt.
[Create-Container.txt](https://github.com/microsoft/navcontainerhelper/files/4903490/Create-Container.txt)

**Comment by freddydk** [comment]
In the error text file it says:

##[error]re-exec error: exit status 1: output: write \\?\C:\ProgramData\Docker\tmp\hcs821450007\Files\Program Files (x86)\ReportBuilder\Microsoft.ReportingServices.GaugeWinControl.dll: **There is not enough space on the disk.**

did you check that?

**Comment by GreatScott000** [comment]
Sorry missed that message.  I saw the nice bright red error and did not look any farther.  What does it mean?
![image](https://user-images.githubusercontent.com/4333956/87200519-a1c3e480-c2f4-11ea-9ff8-abd117d9a195.png)

I did have 6.5GB remaining on the drive so something is consuming a lot of space in the process.

It is progressing now.  I will advise..

**Comment by GreatScott000** [comment]
Well the container is created successfully now but the build step fails.  And the only error I see is the one showing
> Error: No such object: onprem:onprem-14.8.38658.0-gb

![image](https://user-images.githubusercontent.com/4333956/87213763-36dbd300-c31f-11ea-9154-24f208845743.png)

Raw log file.
[Create Build Container Error 2.txt](https://github.com/microsoft/navcontainerhelper/files/4906106/Create.Build.Container.Error.2.txt)

**Comment by GreatScott000** [comment]
With respect to disk space issue
There seems to be an image left over that I can't account for, "busy_franklin" appeared.  There was another one but it seems to have been removed.
![image](https://user-images.githubusercontent.com/4333956/87214014-3e9c7700-c321-11ea-99c0-d5eef090326d.png)

Is there a reason the onprem-5C670948WR image is left running?

**Comment by freddydk** [comment]
busy_franklin comes from an attempt to build an image.
You specify both -artifactUrl and -imagename - means that it will try to build an image for faster access on next runs (as described in the blog posts).
If you run the pipeline again, it probably doesn't fail - because now the image is there.
I know where the error comes from, but it is strange that it becomes an error.
Could you paste your create container part of the pipeline, like:

```
    - task: PowerShell@2
      displayName: 'Create Build Container'
      env:
        Password: $(Password)
        LicenseFile: $(LicenseFile)
      inputs:
        targetType: filePath
        filePath: 'scripts\Create-Container.ps1'
        failOnStderr: true

```

**Comment by GreatScott000** [comment]
Here my "Create Build Container" step
```
    - task: PowerShell@2
      displayName: 'Create Build Container'
      env:
        Password: $(Password)
        LicenseFile: $(LicenseFile)
      inputs:
        targetType: filePath
        filePath: 'scripts\Create-Container.ps1'
        failOnStderr: true
```

A second run over night sees a successful create build container 
![image](https://user-images.githubusercontent.com/4333956/87218497-27727f00-c34b-11ea-8b7c-59c169272b84.png)

**Comment by freddydk** [comment]
Same as mine - and I didn't get the error.
But... - I know exactly where you got the error and I have made a fix in the next containerhelper to ensure that this doesn't happen to other.

**Comment by GreatScott000** [comment]
As always, thanks for the help and the tools.  

Really appreciate your commitment to our community.

**Comment by freddydk** [comment]
Shipped in 0.7.0.14
